### ISSUE TITLE
Goldfish fails to connect to Vault to unwrap the secret-id when the Vault Cluster is in redirect HA mode

### ISSUE THREAD

**Issue by silverbadge** [Caiyeon/goldfish#134]
When the vault cluster is HA and setup to redirect to the leader instead of forwarding, Goldfish fails to follow redirects and results in failing to start due to the fact that it's hitting a standby node. The vault client should follow redirects.

Vault version: 0.7.3
Goldfish version: 0.5.1

**Comment by Caiyeon** [comment]
Does this issue persist if you point goldfish directly at the leader node?

**Comment by Caiyeon** [comment]
This is odd, because presumably the official API should take care of the redirecting. What's the panic error when you launch goldfish?

**Comment by silverbadge** [comment]
My current workaround is to manually point Goldfish to the leader. This does work as long as the leader is not changed. 
Here is a copy of the error:
```
25/07/2017 18:24:59panic: Failed to unwrap provided token, revoke it if possible
25/07/2017 18:24:59Reason:Error making API request.
25/07/2017 18:24:59
25/07/2017 18:24:59URL: PUT https://<REDACTED>/v1/sys/wrapping/unwrap
25/07/2017 18:24:59Code: 400. Errors:
25/07/2017 18:24:59
25/07/2017 18:24:59* error validating wrapping token: Vault is in standby mode
25/07/2017 18:24:59
25/07/2017 18:24:59goroutine 1 [running]:
25/07/2017 18:24:59main.main()
25/07/2017 18:24:59	F:/gopath/src/github.com/caiyeon/goldfish/server.go:80 +0x1dd
```

**Comment by Caiyeon** [comment]
Can you check your vault logs to see if either of these error messages were logged? https://github.com/hashicorp/vault/blob/a362aa97f694407e360c6b1095068276442ccd6b/http/handler.go#L184
https://github.com/hashicorp/vault/blob/a362aa97f694407e360c6b1095068276442ccd6b/http/handler.go#L186

Vault API is setup such that it handles redirects internally. So goldfish should not be responsible for handling redirects in HA mode.

**Comment by Caiyeon** [comment]
Also, if you set your `VAULT_ADDR` to your standby node, and try `vault status` what is the output of the leader address?

E.g.
```bash
$ vault status
Sealed: false
Key Shares: 5
Key Threshold: 3
Unseal Progress: 0
Unseal Nonce:
Version: 0.7.3
Cluster Name:
Cluster ID: 

High-Availability Enabled: true
        Mode: standby
        Leader: http://172.20.20.21:8200
```

The leader address should point to the address of the actual leader node

**Comment by Caiyeon** [comment]
@silverbadge is this still an issue?

**Comment by silverbadge** [comment]
Thanks for the quick response Tony. I can't see any log entries in the vault servers logs that matches the ones you've linked

Our vault cluster is sitting behind a load balancer that excludes sealed vaults nodes. It also excludes the leaders as we want the client to be redirected to the leader and not go through the LB so that we have visibility of the client IP address for approle cidr validation. So, as it stands, the requests are all going to unsealed standby nodes and get redirected to the URL of the Leader correctly.

The Leader address is right when doing vault status on a standby vault node:

```
VAULT_ADDR=https://vaultserver02 vault status -tls-skip-verify
Sealed: false
Key Shares: 5
Key Threshold: 3
Unseal Progress: 0
Unseal Nonce:
Version: 0.7.3
Cluster Name: vault-cluster-e05b5033
Cluster ID: 30dbfa53-e2ca-444c-dc51-5f53ff744348

High-Availability Enabled: true
	Mode: standby
	Leader: https://vaultserver01
```

**Comment by silverbadge** [comment]
Raised an issue on vault for this, https://github.com/hashicorp/vault/issues/3071

**Comment by Caiyeon** [comment]
According to your issue it is a vault API problem independent from goldfish? I will wait for their fix then

**Comment by Caiyeon** [comment]
According to vault issues, this has been fixed in vault server v0.8.0
